src/D2L.CodeStyle.Analyzers/UnsafeStaticsAnalyzer.cs

```diff
@@ -49,6 +49,11 @@ public sealed class UnsafeStaticsAnalyzer : DiagnosticAnalyzer {
         }
 
         private void AnalyzeField( SyntaxNodeAnalysisContext context ) {
+            if( context.Node.SyntaxTree.FilePath.Contains( ".generated" ) ) {

```

Comment:
Is it accurate enough to check for ".generated" anywhere in the path or should it be a component of the path ending in ".generated"? I assume the former is true, but wanted to make sure the other case was considered...